### ISSUE TITLE
Autopopulate hooks screen with "Undeployed" when "Undeployed" or "Exclude" Gear Perf is selected

### ISSUE THREAD

**Issue by jimfellows-NOAA** [nwfsc-fram/pyFieldSoftware#144]
When undeployed is selected on the gear performance screen, auto-populate all five hooks with "Undeployed" when Undeployed or Exclude is selected.  Also deactivate all other gear perfs from being selected.

**Comment by jimfellows-NOAA** [comment]
Got the gear perf. buttons autopopulating, still need to insert "Undeployed" to all five hook entries when Undeployed is pressed.

**Comment by jimfellows-NOAA** [comment]
Have this working with the "Undeployed" button, with dialog asking if you'd like to set all hooks to undeployed for angler.  Figured it might be nice to not automatically reset all the hooks in case button is selected accidentally.

![image](https://user-images.githubusercontent.com/57507677/94623029-2117e300-0268-11eb-8469-bcfc141965a5.png)

@johnharms-NOAA how does the above sound / look?  Does anything need to happen when the "Exclude" button is pressed?  I don't have a good understanding of what that button means.

**Comment by johnharms-NOAA** [comment]
@jimfellows-NOAA  Yes - very good idea to prompt to fill in the hooks rather than just doing it behind the scenes.  And the "Exclude" selection for a drop is made by the HM biologist to denote a situation where the angler on that particular drop is no  longer fishing effectively so that we can exclude the hooks of that particular drop in catch rate calculations. It's doesn't happen often - maybe a few dozen times during a survey year for situations where an angler hangs the bottom immediately and has to do all sorts of violent fishing pole gymnastics just to try and free the line. I believe it's just a simple binary field at the angler-drop level in the SQLite database (at least that's how it ultimately gets stored in the Access database).  We also will exclude, by default, any angler-drop where 3 or more hooks are individually excluded for some reason (usually in the form of "Lost Hook" results. The thinking is that if the majority of hooks on an angler-drop are invalid/excluded, the remaining 2 hooks are likely not a valid representation of the fishing at that particular location. We currently don't have a way of excluding individual hooks in the software right now - it's typically done in Access after the survey.  I'm also not sure that we need that functionality in the s/w either, but others can weigh in on that.

**Comment by jimfellows-NOAA** [comment]
Last two question I think @johnharms-NOAA:

When "Exclude" is seleted, do you also want it to autopopulate the hooks page? 
When "Undeployed" is selected, "Exclude" is autopopulated, and if you unselect Exclude, it will deselect Undeployed. You're also able to have Exclude selected with any other Gear Performance Button (except No Problems).

Might take a bit of testing to figure out exactly how these buttons should behave, but I think they're close to being right.

**Comment by jimfellows-NOAA** [comment]
Closing ticket.  Code merged into develop-hlmatrix.

**Comment by jimfellows-NOAA** [comment]
Can change this, but disabling this dialog if angler hasn't finished fishing yet AKA we can't get to hooks screen yet.  Should simplify code and the app a bit.  Essentially if the hooks screen isn't available, we shouldn't be inserting records there.

**Comment by johnharms-NOAA** [comment]
In general I think not being able to insert records in the hooks screen while the angler is still fishing is a good idea (though we may all need to think about whether there are exceptions to this)...but in the case of an Undeployed gear performance - that usually means that the angler is essentially not taking part in fishing ops that drop, either because the deckhand isn't available when the other anglers are dropping or because of technical problems with the rod/reel.  In the software, however, that angler's timer continues to run, and there's no way to get into the Hooks screen without pressing the time waypoint buttons in order to ultimately get to "At Surface", at which point the Hooks screen becomes available.  I'm thinking that once the HookMatrix person hits a Gear Perf of Undeployed, they should get a pop up message saying something like "Selecting "Undeployed" will void Angler X's drop and set all 5 hooks to "Undeployed" YES/NO - and this should happen without having to press any of the fishing operation waypoints since this angler won't have fished this drop.

**Comment by jimfellows-NOAA** [comment]
Makes sense to me.  So in addition to autopopulating the hooks, hitting YES on the "Do you want to undeploy all hooks?" dialog in the gp screen will also set any of the Start, Begin Fishing, First Bite, Retrieval, and At Surface buttons to null if they don't already have a value, and allow the user to access to the Hooks screen if they need it.  I assume that we'd want to preserve and populated time values, say if the user

**Comment by johnharms-NOAA** [comment]
Seems like the only way that could be an issue would be if someone hit "undeployed" accidentally which I guess could certainly happen. But hopefully the pop up would give the user a moment to think about the situation.  But let's say a user did accidentally select a Gear Perf of Undeployed for a valid drop that did occur and plowed right through the warning pop up message...are you saying that once the time waypoints get set to null and the Hooks get autopopulated as Undeployed, it would then be impossible to try and revert and change the Gear Perf to something else and then try and re-constitute the time waypoints via the Edit function on the Drops screen and then enter new hook results?

Other contingencies - since the drop's gear perf would be Undeployed and all hooks would get autopopulated as Undeployed, I wonder if there should be any sort of a pop-up warning if the user then tries to change the hook results in the Hooks screen to something other than Undeployed?  Like "Angler X's gear performance was Undeployed. Do you really want to change this hook's result from Undeployed?" or something to that effect.

**Comment by jimfellows-NOAA** [comment]
Didn't realize there was an Edit button there.  Alright, well in that case I like the idea of jumping straight to "At Surface" status for the angler and autopopulating the undeployed hooks.  I like the idea of warning the user when changing a hook to something other than undeployed with an undeployed gear type.  I'll work on making those changes.

**Comment by jimfellows-NOAA** [comment]
latest builds starting 2021.0.0+6 should have this revised feature.
